src/core/server/saved_objects/service/lib/errors.ts

```diff
@@ -202,4 +202,23 @@ export class SavedObjectsErrorHelpers {
   public static isGeneralError(error: Error | DecoratedError) {
     return isSavedObjectsClientError(error) && error[code] === CODE_GENERAL_ERROR;
   }
+
+  public static createGenericNotFoundEsUnavailableError(
+    type: string | null = null,
+    id: string | null = null
+  ) {
+    const notFoundError = this.createGenericNotFoundError(type, id);
+    return this.decorateEsUnavailableError(
+      new Error(`${notFoundError.message}`),
+      `x-elastic-product not present or not recognized`
+    );
+  }
+
+  public static isNotFoundEsUnavailableError(error: Error | DecoratedError) {
```

Comment:
Do we need this API? If we want plugins to treat this error as any other ES not found issue, I think it'd be ok to only include the factory function `createGenericNotFoundEsUnavailableError`  so that plugins don't waste time trying to figure out how they should handle this error in a different way.

Reply:
I'm actually using the API within the repository to verify error types [here](https://github.com/elastic/kibana/pull/107301/files?file-filters%5B%5D=.js&file-filters%5B%5D=.ts#diff-01c073903639e29be8562ed0de79fae4084a953304672d1e6bdb3fd7fef06f11R1266). It also ensures we stay consistent with repository error type checks, such as [`isNotFoundError`](https://github.com/elastic/kibana/pull/107301/files?file-filters%5B%5D=.js&file-filters%5B%5D=.ts#diff-7a19c0f1ba03e2b736d8031dc6c5ea8b746cae37d2d81f357577ec34612dbc12R151-R153), with the additional check on the error reason.

Reply:
Right, but as a plugin developer, I'd want to handle each error exposed on `SavedObjectsErrorHelpers.is*` in order to make my code bulletproof. I'd expect that each of these `is*Error` methods to only correspond to one error type and not overlap, but now that is no longer the case since both `isEsUnavailableError` and `isNotFoundEsUnavailableError` could return true on the same error. I'd also expect that each error type would correspond to something different I could do to handle this, however the underlying problem for both of these errors is the same and there's really nothing different I can/should do as a developer for the `isNotFoundEsUnavailableError` case vs the `isEsUnavailable` one.

Are you planning to add similar new error guards for each method as part of https://github.com/elastic/kibana/issues/107343? Personally, I think we should only expose `isEsUnavailable` for all of these situations to signal to plugins that they should handle this error in the same way for every method.

Reply:
> Personally, I think we should only expose isEsUnavailable for all of these situations to signal to plugins that they should handle this error in the same way for every method.

That makes sense. I'll remove `isNotFoundEsUnavailableError`.